### ISSUE TITLE
[MDEP-613] Analyze failed: Unsupported class file major version 55 (Java 11)

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#1045]
**[Mincong Huang](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mincong)** opened **[MDEP-613](https://issues.apache.org/jira/browse/MDEP-613?redirect=false)** and commented

Class file of major version 55 (Java 11) is not yet supported by Maven Dependency Plugin. So when running command `mvn dependency:analysis` on classes created by Java 11, il failed. See `log.txt` for the full log trace.

This is caused by ASM, which does not support major version 55 (Java 11) yet. However, their HEAD contains already the solution, so using the SNAPSHOT version will work. This support will be included in the next release 6.2.

---

**Affects:** 3.1.0

**Attachments:**
- [log.txt](https://issues.apache.org/jira/secure/attachment/12923552/log.txt) (_75.52 kB_)

**Issue Links:**
- [MSHARED-770](https://issues.apache.org/jira/browse/MSHARED-770) Upgrade org.ow2.asm:asm to 7.0
 (_**"depends upon"**_)

**Remote Links:**
- [GitHub Pull Request #8
](https://github.com/apache/maven-dependency-plugin/pull/8)
- [Add support to major version 55 (Java 11)
](https://gitlab.ow2.org/asm/asm/issues/317830)

11 votes, 28 watchers

**Comment by jira-importer** [comment]
**[Anish V. Abraham](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=anishvabraham)** commented

Is there a snapshot version of the plugin where the fix has been applied and can be tested?

**Comment by jira-importer** [comment]
**[Mikaël Geljić](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mgeljic)** commented

I can confirm that since MSHARED-770 / maven-dependency-analyzer 1.11.0-SNAPSHOT, `dependency:analyze` passes all fine (makes sense given the visitors are created with ASM7 support). :)

**Comment by jira-importer** [comment]
**[Henning Schmiedehausen](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=henning)** commented

Any chance to get this wrapped up and released. dependency plugin is one of the few that does not work with Java 11 and it keeps us from upgrading our toolchain completely to Java 11.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Give me a few days cause currently I'm working on maven-dependency-analyzer which is used by maven-dependency-plugin....

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Done in [c756d9edcff039aebe9a0e2d150ad4af49b60b10](https://gitbox.apache.org/repos/asf?p=maven-dependency-plugin.git;a=commitdiff;h=c756d9edcff039aebe9a0e2d150ad4af49b60b10)

**Comment by jira-importer** [comment]
**[Henning Schmiedehausen](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=henning)** commented

Any updates here? This is starting to hurt us with JDK8 creeping ever closer to the -Brexit-EOL date.

**Comment by jira-importer** [comment]
**[Rune Flobakk](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rune)** commented

[hgschmie](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hgschmie): we just override the dependency version for maven-dependency-analyzer, and it works with no problems while we wait for the release of maven-dependency-plugin:

In dependencyManagement:

```xml
<plugin>
  <artifactId>maven-dependency-plugin</artifactId>
  <version>3.1.1</version>
  <dependencies>
    <dependency>
      <groupId>org.apache.maven.shared</groupId>
      <artifactId>maven-dependency-analyzer</artifactId>
      <version>1.11.1</version>
    </dependency>
  </dependencies>
</plugin>
```

**Comment by jira-importer** [comment]
**[Benny Bottema](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bbottema)** commented

Thanks [Rune Flobakk](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rune), that fixed it for me, but this is not ideal. Will there be a new release soon? It's kind of embarrassing to explain to the Gradle comunity.

**Comment by jira-importer** [comment]
**[David J. M. Karlsen](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=davidkarlsen)** commented

It's been a while - could 3.1.2 be released?

**Comment by jira-importer** [comment]
**[Richard Eckart de Castilho](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rec)** commented

+1 for the release - that would be a charm!

**Comment by jira-importer** [comment]
**[Dan Dragut](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dan.dragut)** commented

Included in 3.1.2 release, thank you all! :) (y)
https://blogs.apache.org/maven/entry/apache-maven-dependency-plugin-version2
